### DISCUSSION TITLE
SQL Query based form : problem with update

### DISCUSSION THREAD

**Discussion by NicolasFabre81** [Budibase/budibase#6328] [Help] [answered]
My form is based on a repeater sourced on a sql query that makes a join between 3 tables of my postgreSQL database.

When I want to set up the UPDATE action, a new row is always created in my table, instead of updating the existing one.

I tried to pass the ID as explained on https://github.com/Budibase/budibase/discussions/5739 but no matter it always does insert a new row...

Known issue with postgreSQL ? Workaround ?

**Comment by NicolasFabre81** [comment, marked as answer]
If I change the source of my DP and put a table instead of a join query, it works

**Reply by melohagan** [reply]
Hey @NicolasFabre81 
If you want to do an update with a query, then you would need to use the SQL syntax: `UPDATE table SET ...`

This does mean you would need separate queries for reading and writing. 
Your write query would be executed via the **Execute Query** action, instead of **Save Row**.
